models/progresses.js

```
@@ -146,11 +146,95 @@ const addUserDetailsToProgressDocs = async (progressDocs) => {
+  const result = {
+    successCount: 0,
+    failureCount: 0,
+    successfulRecords: [],
+    failedRecords: []
+  };
+  
+  // First, check for existing progress documents for the current day
+  const existingProgressChecks = await Promise.all(
+    progressDataArray.map(async (progressData) => {
+      try {
+        const { type, taskId, userId } = progressData;
```

ESLint

[error] 'userId' is assigned a value but never used.

(no-unused-vars)